src/ast/passes/semantic_analyser.cpp

```diff
@@ -83,6 +83,46 @@ static std::pair<int64_t, int64_t> getIntTypeRange(const SizedType &ty)
   }
 }
 
+static bool IsValidVarDeclType(const SizedType &ty)
```

Comment:
What is the rationale behind the selection? All the percpu ones cannot be declared? If so, can `ty.NeedsPercpuMap()` be used instead? 

Whatever it is, it should probably be documented. Will help inform the next person who adds/removes a type (hopefully nobody is adding more).

Reply:
Yeah it's all the map aggregation types that can't go into scratch variables. I'll see if they're the same as `NeedsPercpuMap` and document either way.

Reply:
They're not quite the same. stack_mode is not a valid var decl type at the moment - and actually I should probably add 'void' and 'none' for now.